### PR TITLE
allow collection with software events

### LINKED ISSUE #2: Can't create performance event file descriptor: No such file or directory
Testing on current Debian worked fine, testing on another machine (RHEL 8.5) raised this error.
https://github.com/ostash/perfgrind/blob/9b1a75f090cc64866eba7be7ee4368296ea6cba1/pgcollect.c#L307-L314

As `perf record -o perf.data --call-graph dwarf,8192 -z` worked fine I have no idea what to check, do you have any clue why that system call does not work here?

### PR DESCRIPTION
**Pull request by GitMensch**
fixes #2

The result is quite nice, I'm now able to throw pgcollect to some more testing.

### CODE REVIEW

File: pgcollect.c [hunk 1]
```diff
@@ -150,10 +150,12 @@ static void collectExistingMappings(struct PGCollectState* state)
 static void __attribute__((noreturn))
 printUsage()
 {
-  fprintf(stdout, "Usage: %s outfile.pgdata [-F freq] {-p pid | [--] cmd}\n", program_invocation_short_name);
+  fprintf(stdout, "Usage: %s outfile.pgdata [-F freq] [-s] {-p pid | [--] cmd}\n", program_invocation_short_name);
   exit(EXIT_SUCCESS);
 }
 
+static int use_sw_events = 0;
```
Comment (ostash) [inline pgcollect.c:157]: I'd prefer this setting to be part of PGCollectState.

File: pgcollect.c [hunk 2]
```diff
@@ -310,6 +318,9 @@ static int createPerfEvent(const struct PGCollectState* state, pid_t pid, int cp
     perror("Can't create performance event file descriptor");
     if (state->gogoFD != -1)
       close(state->gogoFD);
+    perror("\nHint: check /proc/sys/kernel/perf_event_paranoid");
```
Comment (ostash) [inline pgcollect.c:321]: fputs(message, stderr)

File: pgcollect.c [hunk 3]
```diff
@@ -310,6 +318,9 @@ static int createPerfEvent(const struct PGCollectState* state, pid_t pid, int cp
...
+    if (!use_sw_events)
+      perror("Hint: possibly retry using software events (option -s)");
```
Comment (ostash) [inline pgcollect.c:323]: fputs

**Review by ostash — CHANGES_REQUESTED** [review]
Please change the commit message to something like:

pgcollect: allow to use software events

Virtual machines often lack support of hardware counters, so let us provide some alternative.

**Comment by GitMensch** [comment]
thanks for the review, change in progress (note: I did not find a ChangeLog file, do you care if I add it as part of this PR (separate commit)?

**Comment by ostash** [comment]
I prefer well-written commit messages instead of ChangeLog, no need to add one.
There is however a NEWS file that lists significant changes, but it can be updated later, together with the next release tag.

**Comment by GitMensch** [comment]
I guess this version is better :-)

Question for more possible PRs: I'd send in extended makefile (with clean and check) and configuration for gitpod/vscode, what do you think?

**Comment by GitMensch** [comment]
Gitpod allows online editing (editor based on vscode) and building (based on docker), if configured also "setup code environment with one click" from each branch.

**Comment by ostash** [comment]
Dunno, need to take a close look at that.

**Pull request merged**